### ISSUE TITLE
pypi package install broken if GPUtil is not already install

### ISSUE THREAD

**Issue by bcharlier** [getkeops/keops#9]
the following import in __init__.py should be moved:

https://github.com/getkeops/keops/blob/76578a798cee4885ed1ff7a4d6fc15bf4c54b504/pykeops/__init__.py#L28

no need to any LazyTensor stuff by default.

**Comment by jeanfeydy** [comment]
Hi @bcharlier !
Wow, is this really the source of the broken pypi package? That's yet another crazy error with the setuptools+sphinx suite... I wrote this to let users type

```python
from pykeops import LazyTensor
```

but there must be another way... I'm looking into it.

**Comment by bcharlier** [comment]
The pb is that at installation with get the __version__ string by making 

https://github.com/getkeops/keops/blob/76578a798cee4885ed1ff7a4d6fc15bf4c54b504/setup.py#L9 

But now it implies also the import of LazyTensor (and I guess much much other stuff by domino effect)

**Comment by jeanfeydy** [comment]
I think that https://github.com/getkeops/keops/commit/1e219e187a0fffa7517d981a1f3638283f799a4e does the trick : just put a try-catch-pass around the import, so that the pip installer does not get stuck. I tried it on the test PyPi repository, and it worked in Google Colab :-)

**Comment by bcharlier** [comment]
no, there should ba a better solution. There is no need to load uneeded import as that step!

**Comment by jeanfeydy** [comment]
Mmmm... I really think that importing stuff in the __init__.py is the pythonic way of hiding the folder structure to end users, as detailed [here](https://stackoverflow.com/questions/38130880/in-python-is-it-possible-to-expose-modules-from-subpackages-at-package-level). 
I wouldn't call this an "uneeded import", just a re-indexation of symbolic links... And it works fine, right?

**Comment by bcharlier** [comment]
I have give a try to another solution on the branch  https://github.com/getkeops/keops/commit/968d1a9bc74c8ace5dcb48d9c4bfa9ecf3c4b1c0  by changing the import without doing changes in  the LazyTensor class. It uses now {numpy,torch}tools and pykeops.torch_found which was already defined. 

The main side effect is: to import LaziTensor one should now write

```
from pykeops.numpy import LazyTensor
from pykeops.torch import LazyTensor
```

But the shorter solution :

```
import pykeops.numpy
import pykeops.torch
```

should work also. What do you think ?

**Comment by jeanfeydy** [comment]
Hi Benjamin!
La nuit porte conseil : thinking about it, you're right in the long term, especially if future front-ends (tensorflow?) start allocating large amounts of device memory at import time. Your solution is super clever I think and you must be tired, so please go to bed (it's 2am in Vancouver, right?) : I'll clean the unit tests and the doc today, and you should be able to commit as you wake up.
Thanks :-)

**Comment by jeanfeydy** [comment]
Think that https://github.com/getkeops/keops/commit/d99ec55242132e4ec676c8ba8b107a2f0e87aab7 is good to go. Sphinx is currently running smoothly on Topdyn, and the test PyPi package works on Google Colab: that should be good enough for a 1.1.1 release.

**Comment by bcharlier** [comment]
Fix by #11.
